test/csuite/wt13450_dont_evict_dirty/main.c

```diff
@@ -0,0 +1,280 @@
+
+    /*
+     * Create the thread for cursor->next and wait until we see control point TEST trigger. This
+     * ensures that we can begin the checkpoint after the cursor walking next is in the correct
+     * place in the code.
+     */
+    testutil_check(pthread_create(&next_thread_id, NULL, thread_do_next, opts));
+    /* Wait for our next thread. */
+    CONNECTION_CONTROL_POINT_WAIT(
+      (WT_SESSION_IMPL *)wt_session, WT_CONN_CONTROL_POINT_ID_WT_13450_TEST);
+
+    /* Open a session to run checkpoint. */
+    testutil_check(conn->open_session(conn, NULL, session_open_config, &checkpoint_session));
+
+    /* Open a cursor with which to dirty the tree otherwise checkpoint won't "really" happen. */
+    testutil_check(
+      (ret = checkpoint_session->open_cursor(checkpoint_session, opts->uri, NULL, NULL, &cursor)));
+
+    testutil_check(wt_session->begin_transaction(wt_session, "isolation=snapshot"));
+    /* Dirty the tree. Insert a record at the end. */
+    set_key(cursor, opts->nrecords);
+    set_value(opts, cursor, opts->nrecords);
+    testutil_check(cursor->insert(cursor));
+    testutil_check(wt_session->commit_transaction(wt_session, "commit_timestamp=5"));
+
+    printf("Begin checkpoint\n");
+    /*
+     * Checkpoint the database this will call into __wt_sync_file which will first mark the btree as
+     * syncing, then it will signal the cursor->next thread to continue. There's also a cleverness
+     * about making sure the btree ID matches, we could probably have skipped this? But I think it
+     * demonstrates a nice feature of control points so we choose to have it.
+     *
+     * The reasoning behind why we could have skipped that is that there is only one b-tree that is
+     * relevant and the code paths currently guarantee that it would be the first one up for
+     * checkpoint.
+     */
+    checkpoint_session->checkpoint(checkpoint_session, NULL);
+
+    /* By this stage the program has crashed. */
+    testutil_check(pthread_join(next_thread_id, NULL));
+
+    conn->disable_control_point(conn, WT_CONN_CONTROL_POINT_ID_WT_13450_CKPT);
+    conn->disable_control_point(conn, WT_CONN_CONTROL_POINT_ID_WT_13450_TEST);
+
+    testutil_check(cursor->close(cursor));
+    testutil_check(wt_session->close(wt_session, NULL));
+    testutil_cleanup(opts);
+    return (EXIT_SUCCESS);
+}
+
+/*
+ * thread_do_next --
+ *     Read to trigger pre-fetch.
+ */
+void *
+thread_do_next(void *arg)
+{
+    TEST_OPTS *opts;
+    WT_CONNECTION *conn;
+    WT_CURSOR *cursor;
+    WT_SESSION *wt_session;
+    int ret;
+
+    opts = (TEST_OPTS *)arg;
+    conn = opts->conn;
+
+    printf("Running next thread\n");
+
+    testutil_check(conn->open_session(conn, NULL, session_open_config, &wt_session));
+    testutil_check(wt_session->open_cursor(wt_session, opts->uri, NULL, NULL, &cursor));
+    wt_session->breakpoint(wt_session);
+    /* Wait for the main test thread to get to the control point. */
+    printf("walking cursor next\n");
```

Comment:
I think there may be a better log mechanism than printf for csuite tests can you confirm? If there is please replace the printfs in this file.

Reply:
45 test/csuite programs use printf() or fprintf().
2 test/csuite programs use session->log_printf() which is a part of the normal WT session API.
Thus it is much more common to use printf().
Never-the-less I replaced them with session->log_printf().

Reply:
Log printf will add a message to the write ahead log and is not what should be used, please revert that change.